lib/templates/lab2/lab2.dart

```diff
@@ -1,15 +1,99 @@
 class Book {
-  // your code here
+  String title;
+  String author;
+  int numberOfPages;
+
+  Book({required this.title, required this.author, required this.numberOfPages});
+
+  void bookDetails() {
+    print('Title: $title, Author: $author, Number of Pages: $numberOfPages');
+  }
+}
+
+class FictionBook extends Book {
+  String genre;
+
+  FictionBook({
+    required super.title,
+    required super.author,
+    required super.numberOfPages,
+    required this.genre,
+  });
+
+  @override
+  void bookDetails() {
+    print('Title: $title, Author: $author, Number of Pages: $numberOfPages, Genre: $genre');
+  }
+}
+
+mixin Readable {
+  void read() {
+    print("The book is being read.");
+  }
+}
+
+//In this case i have just created another class (because task doesn't state to use already created ones)
+class ReadableBook with Readable {
+  String title;
+  String author;
+  int numberOfPages;
+
+  ReadableBook({required this.title, required this.author, required this.numberOfPages});
 }
 
-class FictionBook {
-  // your code here
+double calculateArea({required double length, required double width, String shape = "rectangle"}) {
+  if (shape == "circle") {
+    return 3.14 * length * length;
+  } else {
+    return length * width;
```

Comment:
What if not rectangle and not circle?

Reply:
It wasnt stated in the task but i can add something like another else if or when (i forgot how it called in dart)

Reply:
it is okey, just thing that could be improved